src/parser.cpp

```diff
@@ -398,8 +447,17 @@ void Parser::ParserImpl::loadComponent(const ComponentPtr &component, const XmlN
             loadReset(reset, component, childNode);
             component->addReset(reset);
         } else if (childNode->isMathmlElement("math")) {
-            // TODO: copy any namespaces declared in parents into the math element
-            //       so math is a valid subdocument.
+            // Copy any namespaces that do not feature as a namespace definition
+            // of the math node into the math node.
+            auto mathElementDefinedNamespaces = childNode->definedNamespaces();
+            auto possiblyUndefinedNamspaces = traverseTreeForUndefinedNamespaces(childNode);
+            auto undefinedNamespaces = determineMissingNamespaces(possiblyUndefinedNamspaces, mathElementDefinedNamespaces);
+            XmlNamespaceMap::const_iterator it;
+            for (it = undefinedNamespaces.begin(); it != undefinedNamespaces.end(); ++it) {
+                childNode->addNamespaceDefinition(it->second, it->first);
+            }
```

Comment:
This is a big change from the previous state.  Now we look through the math tree and determine which namespaces are used and compares them to what is defined.  The missing namespaces are then added to the root node for the math.